applications/plugins/SofaPython/Binding_Data.cpp

```diff
@@ -185,252 +185,200 @@ PyObject *GetDataValuePython(BaseData* data)
     return PyString_FromString(data->getValueString().c_str());
 }
 
-bool SetDataValuePython(BaseData* data, PyObject* args)
-{
-    // de quel type est args ?
-    bool isInt = PyInt_Check(args);
-    bool isScalar = PyFloat_Check(args);
-    bool isString = PyString_Check(args);
-    bool isList = PyList_Check(args);
+
+static bool SetDataValuePythonVectorLinearSpring(BaseData* data, PyObject* args,
+                                          Data<sofa::helper::vector<LinearSpring<SReal> > >* dataVectorLinearSpring) {
+    const bool isList = PyList_Check(args);
+
     const AbstractTypeInfo *typeinfo = data->getValueTypeInfo(); // info about the data value
-    int rowWidth = (typeinfo && typeinfo->ValidInfo()) ? typeinfo->size() : 1;
-    int nbRows = (typeinfo && typeinfo->ValidInfo()) ? typeinfo->size(data->getValueVoidPtr()) / typeinfo->size() : 1;
+    const bool valid = (typeinfo && typeinfo->ValidInfo());
+        
+    const int rowWidth = valid ? typeinfo->size() : 1;
+    int nbRows = valid ? typeinfo->size(data->getValueVoidPtr()) / typeinfo->size() : 1;
 
-    // horrible special case that needs to be refactored
```

Comment:
basically, I split function body for linearsprings and lists in separate functions